### PR TITLE
[Packaging] Add CI job to build Mariner RPM

### PR DESCRIPTION
**Pull request by mfrw**
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

**Description**

Adds a new build step in the CI to create Mariner RPM.

**Testing Guide**

Tested the script locally.

**History Notes**

Add Build job in CI for Mariner Rpm.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

This PR Depends on:

- #20061
- #20574

### CODE REVIEW

**Comment by jiasli** [comment]
The artifact won't be build because [Build.Reason](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services) doesn't include `PullRequest`.

**Comment by mfrw** [comment]
> The artifact won't be built because [Build.Reason](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services) doesn't include `PullRequest`.

Sure .. Updating :) also this wont work as we are blocked on:

- #20061
The script `mariner.pipeline.sh` will be in once that PR is in

File: azure-pipelines.yml [hunk 1]
```diff
@@ -678,6 +678,28 @@ jobs:
...
+    inputs:
+      targetType: 'filePath'
+      filePath: scripts/release/rpm/mariner_pipeline.sh
+
+
+  - task: PublishPipelineArtifact@0
+    displayName: 'Publish Artifact Mariner: yum'
+    inputs:
+      TargetPath: $(Build.ArtifactStagingDirectory)
+      ArtifactName: yum
```
Comment (jiasli) [inline azure-pipelines.yml:700]:
```suggestion
      ArtifactName: yum-mariner
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -678,6 +678,24 @@ jobs:
...
+  - task: Bash@3
+    displayName: 'Build Rpm Package: Mariner'
+    inputs:
+      targetType: 'filePath'
+      filePath: scripts/release/rpm/pipeline_mariner.sh
+  - task: PublishPipelineArtifact@0
+    displayName: 'Publish Artifact: yum-mariner'
+    inputs:
+      TargetPath: $(Build.ArtifactStagingDirectory)
+      ArtifactName: yum-mariner
```
Comment (jiasli) [inline azure-pipelines.yml:696]: To be honest, I dislike using `yum` and `rpm` interchangeably (as it already is). This makes it difficult to understand for newcomers.

**Review by jiasli — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Comment by mfrw** [comment]
Thank you @jiasli

**Pull request merged**